core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -31,14 +31,18 @@ import org.apache.spark.util.Utils
  * old stages to be resubmitted, such as shuffle map fetch failures.
  */
 @DeveloperApi
-sealed trait TaskEndReason
+sealed trait TaskEndReason {
+  def shouldTaskEndEventuallyFailJob: Boolean
```

Comment:
now that it's a property of `TaskEndReason` it's a little verbose. I would just call this `shouldEventuallyFailJob`